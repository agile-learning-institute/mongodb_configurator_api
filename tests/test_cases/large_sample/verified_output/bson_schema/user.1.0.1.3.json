{
  "description": "A user collection for testing the schema system",
  "bsonType": "object",
  "properties": {
    "_id": {
      "description": "The unique identifier for a user",
      "bsonType": "objectId"
    },
    "user_name": {
      "description": "Username",
      "pattern": "^\\S{1,40}$",
      "bsonType": "string"
    },
    "full_name": {
      "description": "Users Full Name",
      "pattern": "^[^\\t\\n\\r]{0,255}$",
      "bsonType": "string"
    },
    "status": {
      "description": "The status",
      "bsonType": "string",
      "enum": [
        "draft",
        "active",
        "archived"
      ]
    },
    "categories": {
      "description": "A users list of categorized tags",
      "bsonType": "array",
      "items": {
        "description": "A user category",
        "bsonType": "object",
        "properties": {
          "name": {
            "description": "Category Name assigned by the user",
            "pattern": "^\\S{1,40}$",
            "bsonType": "string"
          },
          "category": {
            "description": "The category type assigned by the user",
            "bsonType": "string",
            "enum": [
              "work",
              "personal",
              "project",
              "reference"
            ]
          },
          "tags": {
            "description": "A list of enumerated values assigned by the user",
            "bsonType": "array",
            "items": {
              "bsonType": "string",
              "enum": [
                "urgent",
                "important",
                "normal",
                "low",
                "completed",
                "in_progress",
                "blocked",
                "review"
              ]
            }
          }
        },
        "required": [
          "name",
          "category",
          "tags"
        ],
        "additionalProperties": false
      }
    },
    "preferences": {
      "description": "User preferences and settings",
      "bsonType": "object",
      "properties": {
        "notification_types": {
          "description": "Types of notifications the user wants to receive",
          "bsonType": "array",
          "items": {
            "bsonType": "string",
            "enum": [
              "system",
              "user",
              "content",
              "reminder"
            ]
          }
        },
        "delivery_channels": {
          "description": "Preferred delivery channels for notifications",
          "bsonType": "array",
          "items": {
            "bsonType": "string",
            "enum": [
              "email",
              "sms",
              "push",
              "in_app"
            ]
          }
        },
        "content_tags": {
          "description": "Content tags the user is interested in",
          "bsonType": "array",
          "items": {
            "bsonType": "string",
            "enum": [
              "technology",
              "business",
              "entertainment",
              "education",
              "news"
            ]
          }
        },
        "priority_levels": {
          "description": "Priority levels the user wants to receive",
          "bsonType": "array",
          "items": {
            "bsonType": "string",
            "enum": [
              "critical",
              "high",
              "medium",
              "low"
            ]
          }
        }
      },
      "required": [
        "notification_types",
        "delivery_channels"
      ],
      "additionalProperties": false
    },
    "email": {
      "description": "The person's email address",
      "pattern": "^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$",
      "bsonType": "string"
    },
    "phone": {
      "description": "The person's phone number",
      "pattern": "^\\+1[2-9][0-9]{9}$",
      "bsonType": "string"
    },
    "last_saved": {
      "description": "The last time this document was saved",
      "bsonType": "object",
      "properties": {
        "from_ip": {
          "description": "Http Request remote IP address",
          "pattern": "^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$",
          "bsonType": "string"
        },
        "by_user": {
          "description": "ID Of User",
          "pattern": "^\\S{1,40}$",
          "bsonType": "string"
        },
        "at_time": {
          "description": "The date-time when last updated",
          "bsonType": "date"
        },
        "correlation_id": {
          "description": "The logging correlation ID of the update transaction",
          "pattern": "^\\S{1,40}$",
          "bsonType": "string"
        }
      },
      "additionalProperties": false,
      "required": [
        "from_ip",
        "by_user",
        "at_time",
        "correlation_id"
      ]
    }
  },
  "required": [
    "_id",
    "user_name",
    "status",
    "last_saved"
  ],
  "additionalProperties": false
}