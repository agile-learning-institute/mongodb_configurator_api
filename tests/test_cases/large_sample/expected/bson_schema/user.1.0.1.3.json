{
  "bsonType": "object",
  "required": ["_id", "user_name", "status", "last_saved"],
  "properties": {
    "_id": {
      "bsonType": "objectId",
      "pattern": "^[0-9a-fA-F]{24}$"
    },
    "user_name": {
      "bsonType": "string",
      "pattern": "^[a-zA-Z0-9_-]{4,40}$"
    },
    "full_name": {
      "bsonType": "string",
      "pattern": "^[a-zA-Z0-9\\s.,!?-]{4,255}$"
    },
    "status": {
      "bsonType": "string",
      "enum": ["draft", "active", "archived"]
    },
    "categories": {
      "bsonType": "object",
      "properties": {
        "name": {
          "bsonType": "string",
          "pattern": "^[a-zA-Z0-9_-]{4,40}$"
        },
        "category": {
          "bsonType": "string",
          "enum": ["admin", "user", "guest"]
        },
        "tags": {
          "bsonType": "array",
          "items": {
            "bsonType": "string",
            "enum": ["admin", "user", "guest", "on_hold", "cancelled"]
          }
        }
      }
    },
    "email": {
      "bsonType": "string",
      "pattern": "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$"
    },
    "phone": {
      "bsonType": "string",
      "pattern": "^\\+?[1-9]\\d{1,14}$"
    },
    "last_saved": {
      "bsonType": "object",
      "required": ["from_ip", "by_user", "at_time", "correlation_id"],
      "properties": {
        "from_ip": {
          "bsonType": "string",
          "pattern": "^[a-zA-Z0-9_-]{4,40}$"
        },
        "by_user": {
          "bsonType": "string",
          "pattern": "^[a-zA-Z0-9_-]{4,40}$"
        },
        "at_time": {
          "bsonType": "date"
        },
        "correlation_id": {
          "bsonType": "string",
          "pattern": "^[a-zA-Z0-9_-]{4,40}$"
        }
      }
    }
  }
} 